### PR TITLE
Use callback to avoid async assignation of traits

### PR DESCRIPTION
**Pull request by Quimera996**
https://www.pivotaltracker.com/story/show/150293580

### CODE REVIEW

File: src/js/tracking-wallet.js [hunk 1]
```diff
@@ -576,6 +566,30 @@
     } catch (e) {
       console.error(e);
     }
+  }
+
+
```
Comment (alejandroweb) [inline src/js/tracking-wallet.js]: Remove this blank line please

File: src/js/tracking-wallet.js [hunk 2]
```diff
@@ -538,27 +538,17 @@
       window.analytics.identify(null, params);
     }
   };
-
   /**
      * Start tracking logic
      *
      * @private
      * @name Main#_startTracking
```
Comment (tomasperezv) [inline src/js/tracking-wallet.js]: Update documentation accordingly.

File: src/js/tracking-wallet.js [hunk 3]
```diff
@@ -546,19 +546,10 @@
...
```
Comment (tomasperezv) [inline src/js/tracking-wallet.js]: Update documentation accordingly.

File: src/js/tracking-wallet.js [hunk 4]
```diff
@@ -576,6 +567,29 @@
...
+  var _startTracking = function (initialOptions) {
+    logger.debug('Starting tracking');
+    try {
+      if (
+        typeof initialOptions !== 'undefined' &&
+        typeof initialOptions.userId !== 'undefined'
+      ) {
+        window.analytics.identify(initialOptions.userId, initialOptions.traits, {}, _postInitProcess);
+      } else {
+        window.analytics.identify(null ,null, {}, _postInitProcess);
```
Comment (tomasperezv) [inline src/js/tracking-wallet.js]: The standard way of applying this change is encapsulate what varies (userId and traits) under the if/else leaving out the call to identify, as in:

```javascript
var userId = null;
var traits = null;

if (typeof initialOptions !== 'undefined' && typeof initialOptions.userId !== 'undefined') {
  userId = initialOptions.userId;
  traits = initialOptions.traits;
}

window.analytics.identify(userId, traits, {}, _postInitProcess);
```

**Pull request merged**